bio_tools_oop.py

```diff
@@ -0,0 +1,75 @@
+from abc import ABC, abstractmethod
+
+class BiologicalSequence(ABC):
+
+    def __init__(self, sequence: str):
+        self.sequence = sequence
+ 
+    def __len__(self) -> int:
+        return len(self._sequence)
+
+    @abstractmethod
+    def is_valid(self) -> bool:
+        """
+        Checks if the sequence matches the specified alphabet.
+        Returns:
+            bool: True if the sequence is correct, otherwise False.
+        """
+        pass
+
+    def __getitem__(self, key):
+        return self.sequence[key]
+    
+    def __str__(self) -> str:
+        return self.sequence
+    
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}('{self.sequence}')"
+
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    complement_map = {}
+
+    def is_valid(self) -> bool:
+        return all(nucleotide in self.complement_map for nucleotide in self.sequence)
+
+    def complement(self):
+        return ''.join(self.complement_map[nucleotide] for nucleotide in self.sequence)
```

Comment:
Получается метод возвращает строку, хотя была нуклеиновая кислота. Это можно было бы сделать так:
```suggestion
        return type(self)(''.join(self.complement_map[nucleotide] for nucleotide in self.sequence))
```